src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -310,7 +313,7 @@ export class StudyViewPageStore {
         };
     }
 
-    private clinicalDataCache:ClinicalDataCache = new ClinicalDataCache()
+    private clinicalDataPromises:{[id:string]:MobxPromise<any>} = {};
```

Comment:
i would think the typing on that is pretty important.  should just be like, ClinicalAttribute[] ??

Reply:
This is a mix proxy, but definitely can be updated